jenkins/manila-functional-tests/20-install-openstack.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash -xue
+
+# Set working directory to where the fabfile resides.
+SCRIPT_DIR=$(dirname ${0})
+cd ${SCRIPT_DIR}
+
+# Remove any deployed resources on error.
+trap "set +u && source heat-venv/bin/activate && fab destroy" ERR
```

Comment:
This solution for teardown is a bit ugly. An alternative I have thought of is to handle this teardown in the short script which initiate the job on the jenkins side. This has the downside of starting jobs in different ways, which might be confusing.